### DISCUSSION TITLE
Invalid instructions when reading from a rust x64 binary

### DISCUSSION THREAD

**Discussion by RadonCoding** [icedland/iced#321] [Q&A] [answered]
I'm reading the .text section of a rust x64 binary and getting the following invalid instructions.

Here is my code https://paste.ofcode.org/6rTPNCdgRzRZN6VdVvfk98

```
41 FE FF
FE FB
DD FC FF
FF EE
82 FD
C6 FD
FF 38
8D FE
FF ED
4F FF FF
C5 FD FF FF
82 FE
60 FD
FF DE
8F FF FF FF
FF DA
F0 FD
FF D8
FF EF
44 FF FF
FF BD
FF B8
EA FC
27 FC
3F FC
17 FD
2F FD
8F FD FF FF
07 FE
1F FE
37 FE
4F FE FF
FF 3F
8C FF
FF B9
FF FB
0E FD
FF 7B
FF 3B
CE FD
FF DB
DB FD
C5 FC FF FF
67 FE FF
C7 FA
D6 FA
FF FC
FF DF
FE FE
FF 7E
FF BC
FF 7C
FE FF
8E FF
FF FF
```

**Comment by RadonCoding** [comment]
My goal is to decode the instructions then modify them and encode

**Reply by wtfsck** [reply, marked as answer]
Compilers add data to the code section too so you can't just assume everything is valid code.

You have to figure out which bytes are code and which aren't. Start at some known locations that you know are code, then disassemble that code, follow branches etc, to find new code (see `Instruction::flow_control()` method). Whatever hasn't been decoded could still be code but it's e.g. referenced from some data. Or it's just data.

**Reply by RadonCoding** [reply]
thanks
